### PR TITLE
Vendor in latest containers/image and containers/storage

### PR DESCRIPTION
**Pull request by umohnani8**
Changes to paramters in functions.
Change to the error returned by the oci and oci-archive transport.

Addresses https://github.com/projectatomic/buildah/issues/558

Signed-off-by: umohnani8 <umohnani@redhat.com>

### CODE REVIEW

File: tests/imgtype.go [hunk 1]
```diff
@@ -111,22 +112,22 @@ func main() {
 			ref = ref2
 		}
 
-		img, err := ref.NewImage(systemContext)
+		img, err := ref.NewImage(context.TODO(), systemContext)
```
Comment (mtrmac) [inline tests/imgtype.go]: (In tests it’s OK to call `context.Background()`, i.e. without having them in the “to be fixed later” list, maybe unless you are planning to actually test the cancelation mechanism.)

File: cmd/buildah/common.go [hunk 2]
```diff
@@ -99,7 +100,7 @@ func getDateAndDigestAndSize(image storage.Image, store storage.Store) (time.Tim
 	if err != nil {
 		return created, "", -1, err
 	}
-	img, err := storeRef.NewImage(nil)
+	img, err := storeRef.NewImage(context.TODO(), nil)
```
Comment (nalind) [inline cmd/buildah/common.go]: Go ahead and add a local function to `common.go` that'll return a context for use in the binary, have it return `context.TODO()`, and use it instead.  It'll be a smaller change to not use TODO later.

File: image.go [hunk 3]
```diff
@@ -65,12 +65,12 @@ type containerImageSource struct {
...
 
-func (i *containerImageRef) NewImage(sc *types.SystemContext) (types.ImageCloser, error) {
-	src, err := i.NewImageSource(sc)
+func (i *containerImageRef) NewImage(ctx context.Context, sc *types.SystemContext) (types.ImageCloser, error) {
+	src, err := i.NewImageSource(context.TODO(), sc)
 	if err != nil {
 		return nil, err
 	}
-	return image.FromSource(sc, src)
+	return image.FromSource(context.TODO(), sc, src)
```
[the remark this answers is not in the document]
Comment (umohnani8) [reply image.go]: fixed

File: tests/imgtype.go (hunk 1 above)
Comment (umohnani8) [reply tests/imgtype.go]: calling context.Background() now.

File: cmd/buildah/common.go (hunk 2 above)
Comment (umohnani8) [reply cmd/buildah/common.go]: fixed. PTAL

**Comment by rhatdan** [comment]
@nalind @mtrmac Ok to merge?

**Comment by nalind** [comment]
LGTM.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**